xbmc/guilib/GUIIncludes.cpp

```diff
@@ -18,71 +18,37 @@
 #include "utils/XMLUtils.h"
 #include "utils/log.h"
 
+#include <algorithm>
+
 using namespace KODI::GUILIB;
 
-CGUIIncludes::CGUIIncludes()
-{
-  m_constantAttributes.insert("x");
-  m_constantAttributes.insert("y");
-  m_constantAttributes.insert("width");
-  m_constantAttributes.insert("height");
-  m_constantAttributes.insert("center");
-  m_constantAttributes.insert("max");
-  m_constantAttributes.insert("min");
-  m_constantAttributes.insert("w");
-  m_constantAttributes.insert("h");
-  m_constantAttributes.insert("time");
-  m_constantAttributes.insert("acceleration");
-  m_constantAttributes.insert("delay");
-  m_constantAttributes.insert("start");
-  m_constantAttributes.insert("end");
-  m_constantAttributes.insert("center");
-  m_constantAttributes.insert("border");
-  m_constantAttributes.insert("repeat");
-
-  m_constantNodes.insert("posx");
-  m_constantNodes.insert("posy");
-  m_constantNodes.insert("left");
-  m_constantNodes.insert("centerleft");
-  m_constantNodes.insert("right");
-  m_constantNodes.insert("centerright");
-  m_constantNodes.insert("top");
-  m_constantNodes.insert("centertop");
-  m_constantNodes.insert("bottom");
-  m_constantNodes.insert("centerbottom");
-  m_constantNodes.insert("width");
-  m_constantNodes.insert("height");
-  m_constantNodes.insert("offsetx");
-  m_constantNodes.insert("offsety");
-  m_constantNodes.insert("textoffsetx");
-  m_constantNodes.insert("textoffsety");
-  m_constantNodes.insert("textwidth");
-  m_constantNodes.insert("spinposx");
-  m_constantNodes.insert("spinposy");
-  m_constantNodes.insert("spinwidth");
-  m_constantNodes.insert("spinheight");
-  m_constantNodes.insert("radioposx");
-  m_constantNodes.insert("radioposy");
-  m_constantNodes.insert("radiowidth");
-  m_constantNodes.insert("radioheight");
-  m_constantNodes.insert("sliderwidth");
-  m_constantNodes.insert("sliderheight");
-  m_constantNodes.insert("itemgap");
-  m_constantNodes.insert("bordersize");
-  m_constantNodes.insert("timeperimage");
-  m_constantNodes.insert("fadetime");
-  m_constantNodes.insert("pauseatend");
-  m_constantNodes.insert("depth");
-  m_constantNodes.insert("movement");
-  m_constantNodes.insert("focusposition");
-
-  m_expressionAttributes.insert("condition");
-
-  m_expressionNodes.insert("visible");
-  m_expressionNodes.insert("enable");
-  m_expressionNodes.insert("usealttexture");
-  m_expressionNodes.insert("selected");
-}
+static constexpr std::array<std::string_view, 17> constantAttributes = {
+    "x",     "y",   "width",  "height", "center",       "max",
+    "min",   "w",   "h",      "time",   "acceleration", "delay",
+    "start", "end", "center", "border", "repeat",
+};
```

Comment:
I'm not opposed to switch to `std::array`, but I think using the `std::range::find` in the checks hurts readability a little. I suggest to add a small function that makes the intention clearer:
```cpp
static constexpr std::array<std::string_view, 17> CONSTANT_ATTRIBUTES = {
    "x",     "y",   "width",  "height", "center",       "max",
    "min",   "w",   "h",      "time",   "acceleration", "delay",
    "start", "end", "center", "border", "repeat",
};
static constexpr bool IsConstantAttribtute(const TiXmlAttribute& attribute)
{
  return std::ranges::find(CONSTANT_ATTRIBUTES, attribute.Name()) != CONSTANT_ATTRIBUTES.end();
}
```

Same for the other arrays below.

Reply:
unfortunately std::range::contains seems to be C++23, but you could model after that for easy removal later.

Reply:
Yes, that's also a good solution.